segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -815,6 +821,62 @@ public void readTableEntries(WireCommands.ReadTableEntries readTableEntries) {
+        final String segment = readTableEntriesDelta.getSegment();
+        final String operation = "readTableEntriesDelta";
+
+        if (!verifyToken(segment, readTableEntriesDelta.getRequestId(), readTableEntriesDelta.getDelegationToken(), operation)) {
+            return;
+        }
+
+        final int suggestedEntryCount = readTableEntriesDelta.getSuggestedEntryCount();
+        final long fromPosition = readTableEntriesDelta.getFromPosition();
+
+        log.info(readTableEntriesDelta.getRequestId(), "Fetching keys from {}.", readTableEntriesDelta);
+
+        val timer = new Timer();
+        val result = new DeltaIteratorResult<BufferView, Map.Entry<WireCommands.TableKey, WireCommands.TableValue>, DeltaIteratorState>(
+                segment.getBytes().length + WireCommands.TableEntriesRead.HEADER_BYTES);
+        tableStore.entryDeltaIterator(segment, fromPosition, TIMEOUT)
+                .thenCompose(itr -> itr.collectRemaining(
+                        e -> {
+                            if (result.getItemCount() >= suggestedEntryCount || result.getSizeBytes() >= MAX_READ_SIZE) {
+                                return  false;
+                            }
+                            TableEntry entry = e.getEntries().iterator().next();
+                            DeltaIteratorState state = DeltaIteratorState.deserialize(e.getState().array());
+                            // Store all TableEntries.
+                            val k = new WireCommands.TableKey(toByteBuf(entry.getKey().getKey()), entry.getKey().getVersion());
+                            val v = new WireCommands.TableValue(toByteBuf(entry.getValue()));
+                            if (state.isDeletionRecord()) {
+                                result.remove(entry.getKey().getKey(), k.size() + v.size());
+                            } else {
+                                Map.Entry<WireCommands.TableKey, WireCommands.TableValue> old = result.getItem(entry.getKey().getKey());
+                                if (old != null && old.getKey().getKeyVersion() < entry.getKey().getVersion()) {
+                                    int sizeBytes = (k.size() + v.size()) - (old.getKey().size() + old.getValue().size());
+                                    result.add(entry.getKey().getKey(), new AbstractMap.SimpleImmutableEntry<>(k, v), sizeBytes);
+                                } else {
+                                    result.add(entry.getKey().getKey(), new AbstractMap.SimpleImmutableEntry<>(k, v), k.size() + v.size());
+                                }
+                            }
+                            result.setState(state);
+                            // Update total read data.
+                            return true;
+                        }))
+                .thenAccept(v -> {
+                    connection.send(new WireCommands.TableEntriesDeltaRead(
```

Comment:
Can we add a `log.debug` here (just like the other iterator methods) to indicate that we are sending something back. Feel free to follow the message pattern used for the other iterators.